File: shoppingCart/src/main/java/com/example/shoppingCart/models/Customer.java
package com.example.shoppingCart.models;

import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String name;

    @OneToMany(mappedBy = "customer")
    private List<Orde_r> orders;

    @OneToOne(mappedBy = "customer")
    private Cart cart;



}

File: Submission/angular/shopping-cart/src/app/models/Customer.ts
import { IOrder } from './Order';
import { ICart } from './Cart';

export interface ICustomer {
    id: number;
    name: string;
    orders: IOrder[];
    cart: ICart;
}

export class Customer {
    id: number = 0;
    name: string;
    orders: IOrder[];
    cart: ICart;
}